### ISSUE TITLE
string -> conversion warning / panic on git push

### ISSUE THREAD

**Issue by mboersma** [deis/builder#204]
This was in my logs during a `git push deis master` operation:
```console
[info] Channel type: session
Key='LANG', Value='en_US.UTF-8'
[debug] receiving git repo name: '/united-addendum.git', operation: git-receive-pack, fingerprint: e1:c3:b0:7f:91:d0:47:00:23:19:25:ed:34:6a:c2:9d, user: matt
errorPanic in safely.Go: interface conversion: string is not error: missing method Error
[info] Checking closer.
```

**Comment by arschles** [comment]
@mboersma good to know. I've been trying to repro this and can't. I'm not surprised that it happened, though, because we do type assertions & conversions all over the place. #175 will remove many, if not all of them, although that's a long way out.

**Comment by sstarcher** [comment]
I just got this after setting up Deisv2 beta 2.  Completely fresh setup.
```
deis register
deis git:remote -a app
git push deis
```

```
[info] Checking closer.
[info] Accepted connection.
[debug] Starting ssh authentication
[info] Key accepted for user sstarcher.
[info] Channel type: session
Key='LANG', Value='en_US.UTF-8'
Key='LC_CTYPE', Value='en_US.UTF-8'
[debug] receiving git repo name: '/grafana.git', operation: git-receive-pack, fingerprint: 66:2a:8b:df:32:63:4b:b9:20:5b:f6:3d:13:96:dc:4e, user: sstarcher
errorPanic in safely.Go: interface conversion: string is not error: missing method Error
```

**Comment by arschles** [comment]
@sstarcher do you have any relevant builder pod logs?

**Comment by sstarcher** [comment]
@arschles those are the builders logs.  Here are the rest of the logs nothing interesting though.

```
2016/04/26 20:06:47 Starting health check server on port 8092
2016/04/26 20:06:47 Starting deleted app cleaner
2016/04/26 20:06:47 Starting SSH server on 0.0.0.0:2223
[debug] Generating ssh keys for sshd
[debug] Parsing ssh host keys
[info] Parsed host key /etc/ssh/ssh_host_rsa_key.
[info] Parsed host key /etc/ssh/ssh_host_dsa_key.
[info] Parsed host key /etc/ssh/ssh_host_ecdsa_key.
[debug] Name: EXTERNAL_PORT, Val: 2223
[info] Added hostkey.
[info] Added hostkey.
[info] Added hostkey.
[info] Listening on 0.0.0.0:2223
[info] Accepting new connections.
[info] Checking closer.
[info] Builder is running.
[info] Checking closer.
[info] Accepted connection.
[debug] Starting ssh authentication
[info] Key accepted for user sstarcher.
[info] Channel type: session
Key='LANG', Value='en_US.UTF-8'
Key='LC_CTYPE', Value='en_US.UTF-8'
[debug] receiving git repo name: '/grafana.git', operation: git-receive-pack, fingerprint: 66:2a:8b:df:32:63:4b:b9:20:5b:f6:3d:13:96:dc:4e, user: sstarcher
errorPanic in safely.Go: interface conversion: string is not error: missing method Error
```

**Comment by sstarcher** [comment]
If I do a second push
```
[info] Checking closer.
[info] Accepted connection.
[debug] Starting ssh authentication
[info] Key accepted for user sstarcher.
[info] Channel type: session
Key='LANG', Value='en_US.UTF-8'
Key='LC_CTYPE', Value='en_US.UTF-8'
[error] Another git push is ongoing
```

**Comment by sstarcher** [comment]
@arschles I just realized why it's doing that...
I manually added the git remote the app was never actually created.  So the actual bug is the error message.

With the new setup all I did was
```deis git:remote -a grafana```
instead of 
```
deis create grafana
```

**Comment by arschles** [comment]
@sstarcher thanks for all of that. Right now, the workaround to fix the `Another git push is ongoing` issue is to restart the builder pod. We had an issue for it but it may have been inadvertently closed, so I've created #309 to track it

Also, we're tracking the error message work in #91

**Comment by arschles** [comment]
Moving this into beta4

**Comment by arschles** [comment]
This should be fixed by #316

**Comment by arschles** [comment]
As of #316, this issue should no longer arise. Closing
